scripts/cmake/config/default_ex.cmake

```diff
@@ -0,0 +1,129 @@
+#
+# Bareflank Hypervisor
+# Copyright (C) 2015 Assured Information Security, Inc.
+#
+# This library is free software; you can redistribute it and/or
+# modify it under the terms of the GNU Lesser General Public
+# License as published by the Free Software Foundation; either
+# version 2.1 of the License, or (at your option) any later version.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public
+# License along with this library; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+
+# ------------------------------------------------------------------------------
+# README
+# ------------------------------------------------------------------------------
+
+# This file defines all configurable cmake variables available to Barefank
+# VMM extensions, set to their default value. These configuration are NOT
+# available to base Bareflank hypervisor projects. VMM extension projects
+# can use both these configurations PLUS the base hypervisor configurations.
+
+# Each configuration here indicates the Bareflank suggested convention as
+# a default value. The goal is to encourage a "convention over configuration"
+# mindset for extension developers. If extension developers choose not follow
+# a suggested convention, they can override a configuration using two methods:
+#
+# 1) Specify each configuration as CMAKE_ARGS to the vmm_extension() macro. Ex:
+#
+#     vmm_extension(
+#           extension_name
+#           CMAKE_ARGS -DVMM_EX_BUILD_RULES=/path/to/build_rules.cmake
+#     )
+#
+# 2) Override a configuration in the extension's CMakeLists.txt file BEFORE
+#    calling include(${VMM_EXTENSION}). Ex:
+#
+#     set(VMM_EX_BUILD_RULES "/path/to/build_rules.cmake")
+#     set(VMM_EX_UNITTEST_PATH "/path/to/the/unit/tests")
+#     include(${VMM_EXTENSION})
+#
+
+# ------------------------------------------------------------------------------
+# Source tree structure
+# ------------------------------------------------------------------------------
+
+add_config(
+    CONFIG_NAME VMM_EX_TOP_DIR
+    CONFIG_TYPE PATH
+    DEFAULT_VAL ${CMAKE_CURRENT_SOURCE_DIR}
+    DESCRIPTION "Path to the top directory for this VMM extension"
+)
+
+add_config(
```

Comment:
Are these optional? If not they should be, and our example should be the simplest possible config with options commented out with comments on how to use them if desired. Most of these seem like things we should be able to "infer" based on our "preferred" directory layout

Reply:
The configs in this file are what get passed into the extension by the base build system, set to their default values. Extension writers can override these values in their extension-specific configuration files if desired. The configs here are exactly what you described: things we should be able to "infer" based on our "preferred" directory layout

Reply:
After I read your comment again, I think you're viewing this file as the "template" config file for an extension. That was not the intent here, this is meant to be setup by the base build system and "passed into" the extension. Extensions then define their own config files to define more values, or override the ones here

Reply:
ok yeah, so I think the issue for me was the "_ex", as I read that as example. Please rename to "_extension" and you should be good here